plat/arm/board/fvp/aarch64/fvp_helpers.S

```diff
@@ -205,9 +203,9 @@ func plat_arm_calc_core_pos
 	ubfx	x2, x3, #MPIDR_AFF2_SHIFT, #MPIDR_AFFINITY_BITS
 
 	/* Compute linear position */
-	mov	x4, #FVP_MAX_PE_PER_CPU
-	madd	x0, x1, x4, x0
-	mov	x5, #FVP_MAX_CPUS_PER_CLUSTER
-	madd	x0, x2, x5, x0
+	mov	x3, #FVP_MAX_PE_PER_CPU
+	madd	x1, x2, x3, x1
```

Comment:
This line corresponds to x1=  (cluster_id * FVP_MAX_PE_PER_CPU) + CPU_ID
and the next line x0 = (x1 * FVP_MAX_CPUS_PER_CLUSTER) + thread ID

Shouldn't it be 
```
	mov	x3, #FVP_MAX_CPUS_PER_CLUSTER
	madd	x1, x2, x3, x1
	mov	x3, #FVP_MAX_PE_PER_CPU
	madd	x0, x1, x3, x0
	ret
```

Reply:
@soby-mathew 
Yes your suggestion looks correct.

Reply:
Yes. You are right. That is my mistake.
I have updated the code. Please review it again. Thanks!